<launch>
    
    <include file="$(find rov_tf_tree)/launch/brov_tf_tree.launch"/>
    <param name="use_sim_time" value="false" /><!--use <two lines>clock bellow if true-->
    <node pkg="rosbag" type="play" name="data_play" output="screen" args="/home/catkin_ws/src/bagFiles/Data_2023-11-17-perfect-square.bag"/>
   
    <arg name="camera_start_delay" default="0.0"/>

    <!--Yolo object detection script-->
    <!--node pkg="recognition" name="yolo_detection" type="ros_recognition_yolo.py" output="screen" launch-prefix="bash -c 'sleep $(arg camera_start_delay); $0 $@' "></node-->

    <node pkg="rov_tf_tree" name="plane_camera" type="color2.py" output="screen" launch-prefix="bash -c 'sleep $(arg camera_start_delay); $0 $@' "></node>

    
    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" clear_params="true" output="screen">
        <param name="frequency" value="20"/>
        <!--param name="sensor_timeout" value="0.01"/-->

        <param name="predict_to_current_time" value="true"/>

        <!--param name="map_frame" value="map"/-->
        <param name="odom_frame" value="world"/>
        <param name="base_link_frame" value="Rov_frame"/>
        <param name="world_frame" value="world"/>

        <param name="imu0" value="/BlueRov2/imu"/>
        <!--param name="imu0" value="/mavros/imu/data"/-->
        <!--param name="imu0_queue_size" value="1000"/-->
        <param name="imu0_relative" value="false"/>


        <rosparam param="initial_state">[0.69,0.44,0,
                                        0,0,0,
                                        0,0,0,
                                        0,0,0,
                                        0,0,0]</rosparam>

        <param name="use_control" value="false"/>

        <rosparam param="imu0_config">[false, false, false,
                                 true, true,  true,
                                 false, false, false,
                                 true, true, true,
                                 false,  false,  false]</rosparam>

        <param name="imu0_remove_gravitational_acceleration" value="true"/>
        <param name="dynamic_process_noise_covariance" value="false"/>
        <param name="imu0_differential" value="false"/>
       

	    <param name="odom0" value="/BlueRov2/real_coord"/>
        <rosparam param="odom0_config">[true,true,false,
                                        false,false,false,
                                        false,false,false,
                                        false,false,false,
                                        false,false,false]</rosparam>

        <param name="odom1" value="/BlueRov2/odom/depth"/>
        <rosparam param="odom1_config">[false,false,true,
                                        false,false,false,
                                        false,false,false,
                                        false,false,false,
                                        false,false,false]</rosparam>
        
        <param name="reset_on_time_jump" value="true"/>
        <rosparam param="process_noise_covariance">
                            [0.1, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0.1, 0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0.06, 0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0.03, 0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0.03, 0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0.06, 0,     0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0.025, 0,     0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0.025, 0,    0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0.04, 0,    0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0.01, 0,    0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0.01, 0,    0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0.02, 0,    0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0.01, 0,    0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0.01, 0,
                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0.015]</rosparam>
    </node>
    

</launch>
